pypdf/generic/_base.py

```diff
@@ -282,9 +282,27 @@ def indirect_reference(self) -> "IndirectObject":  # type: ignore[override]
 
     def get_object(self) -> Optional["PdfObject"]:
         obj = self.pdf.get_object(self)
-        if obj is None:
-            return None
-        return obj.get_object()
+        return obj
+
+    def __deepcopy__(self, memo: Any) -> "IndirectObject":
+        return IndirectObject(self.idnum, self.generation, self.pdf)
+
+    def __getattr__(self, name: str) -> Any:
+        # Attribute not found in object: look in pointed object
+        try:
+            return getattr(self.get_object(), name)
```

Comment:
As with the initial implementation of the other PR, this will possibly exceed the recursion depth if I am not mistaken.